### PR TITLE
Implement ItemMergeEvent, ExpBottleEvent, CreeperPowerEvent (#922)

### PR DESCRIPTION
**Pull request by VaiTon**

### CODE REVIEW

File: src/main/java/net/glowstone/entity/monster/GlowCreeper.java [hunk 1]
```diff
@@ -59,7 +63,15 @@ protected Sound getHurtSound() {
     public void damage(double amount, Entity source, DamageCause cause) {
         super.damage(amount, source, cause);
         if (DamageCause.LIGHTNING.equals(cause)) {
```
Comment (aramperes) [inline src/main/java/net/glowstone/entity/monster/GlowCreeper.java]: Off-topic, but could you add a check for `!isPowered()` in this if statement?

File: src/main/java/net/glowstone/entity/projectile/GlowThrownExpBottle.java [hunk 2]
```diff
@@ -28,6 +31,8 @@ public void collide(LivingEntity entity) {
 
     private void spawnOrb() {
         int xp = ThreadLocalRandom.current().nextInt(9) + 3;
+        ExpBottleEvent event = EventFactory.getInstance()
+                .callEvent(new ExpBottleEvent(this, xp));
```
Comment (aramperes) [inline src/main/java/net/glowstone/entity/projectile/GlowThrownExpBottle.java:35]: Set the `xp` variable to the result of `event.getExperience()`

**Review by aramperes — CHANGES_REQUESTED** [review]

File: src/main/java/net/glowstone/entity/projectile/GlowThrownExpBottle.java (hunk 2 above)
Comment (VaiTon) [reply src/main/java/net/glowstone/entity/projectile/GlowThrownExpBottle.java:35]: But `xp` has to be calculated before calling the new event

Comment (aramperes) [reply src/main/java/net/glowstone/entity/projectile/GlowThrownExpBottle.java:35]: I mean after the event is fired. The event object is mutable by plugins, which can use setExperience to change the behaviour. 

So, after event is called, use `xp = event.getExperience();` and keep the rest as is.

**Review by aramperes — APPROVED** [review]

**Comment by aramperes** [comment]
Thank you for your contribution!

**Pull request merged**